### PR TITLE
feat: Add new m365 icon, add brand icons page to docs, and refactor code

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR adds the following:
* Adds test, start, and build commands.
* Adds a new section to the docs to visualize the brand icons.
* Refactors cdn url usage and updates to latest cdn link.
* Adds new m365 icons.

Fixes https://github.com/microsoft/fluentui/issues/25928

### CODE REVIEW

File: src/sass/variables/_Icon.Variables.scss [hunk 1]
```diff
@@ -1,7 +1,9 @@
 // Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE in the project root for license information.
 
 // Icon directory and paths
-$ms-icon-cdn-path: 'https://static2.sharepointonline.com/files/fabric/assets/icons' !default;
+$ms-cdn-build-date: '20221201.001';
+$ms-cdn-path: 'https://res-1.cdn.office.net/files/fabric-cdn-prod_#{$ms-cdn-build-date}/assets' !default;
```
Comment (micahgodbolt) [inline src/sass/variables/_Icon.Variables.scss:5]: lets move these out to another file. i think this one should only be about icons in the same way that BrandIcons.Variables is only about BrandIcons

maybe keep it simple with a _cdn.variables.scss file

File: src/sass/_BrandIcon.scss [hunk 2]
```diff
@@ -22,87 +23,87 @@
 // a future release. Until then, they are mapped to 'one' and 'xlsx'.
 
 .ms-BrandIcon--onepkg.ms-BrandIcon--icon16 {
-  background-image: url(https://spoprod-a.akamaihd.net/files/fabric/assets/item-types-fluent/16/one.png);
+  background-image: url('#{$ms-cdn-path}/item-types-fluent/16/one.png');
```
Comment (Jahnp) [inline src/sass/_BrandIcon.scss:26]: As long as you're changing these, you should be able to replace `item-types-fluent` with `item-types`. That's the more up to date and maintained assets directory.

**Review by Jahnp — APPROVED** [review]
Thanks for this @sopranopillow! Looks great.

File: src/sass/_BrandIcon.scss (hunk 2 above)
Comment (sopranopillow) [reply src/sass/_BrandIcon.scss:26]: will make the change!

**Pull request merged**